src/dst/test/cpp/TestBalancer.cc

```diff
@@ -0,0 +1,104 @@
+#define BOOST_TEST_MODULE TestBalancer
+
+#include <boost/asio.hpp>
+#include <boost/test/included/unit_test.hpp>
+#include <boost/thread/thread.hpp>
+#include <string>
+
+#include "HelperCallBack.h"
+#include "LoadBalancer.h"
+#include "dst/BroadcastJobDescription.h"
+#include "dst/Distributed.h"
+#include "dst/JobMessage.h"
+#include "utl/Logger.h"
+
+using namespace dst;
+using namespace std;
+
+BOOST_AUTO_TEST_SUITE(test_suite)
+
+BOOST_AUTO_TEST_CASE(test_default)
+{
+  utl::Logger* logger = new utl::Logger();
+  Distributed* dist = new Distributed(logger);
+  string local_ip = "127.0.0.1";
+  unsigned short balancer_port = 5555;
+  unsigned short worker_port_1 = 5556;
+  unsigned short worker_port_2 = 5557;
+  unsigned short worker_port_3 = 5558;
+  unsigned short worker_port_4 = 5559;
+  asio::io_service io_service;
+  LoadBalancer* balancer = new LoadBalancer(
+      dist, io_service, logger, local_ip.c_str(), "", balancer_port);
+
+  // Checking simple interface functions
+  balancer->addWorker(local_ip, worker_port_1);
+  balancer->addWorker(local_ip, worker_port_2);
+  asio::ip::address address;
+  unsigned short port;
+
+  balancer->getNextWorker(address, port);
+  BOOST_TEST(address.to_string() == local_ip);
+  BOOST_TEST(port == worker_port_1);
+  balancer->getNextWorker(address, port);
+  BOOST_TEST(address.to_string() == local_ip);
+  BOOST_TEST(port == worker_port_2);
+  balancer->updateWorker(asio::ip::address::from_string(local_ip),
+                         worker_port_2);
+
```

Comment:
I don't think we are testing updateWorker's effect here because we don't use getNextWorker after that to confirm that it would return worker_port_2

Reply:
I was testing it implicitly assuming the balancer would not error when it is selected as the next worker and would as it is not running. Will do it explicitly by calling getNextWorker() here.